src/main/java/io/confluent/connect/jdbc/sink/PreparedStatementBinder.java

```diff
@@ -85,28 +137,34 @@ public void bindRecord(SinkRecord record) throws SQLException {
       case RECORD_VALUE: {
         for (String fieldName : fieldsMetadata.keyFieldNames) {
           final Field field = schemaPair.valueSchema.field(fieldName);
-          bindField(index++, field.schema().type(), ((Struct) record.value()).get(field));
+          bindField(index++, field.schema(), ((Struct) record.value()).get(field));
         }
       }
       break;
     }
 
     for (final String fieldName : fieldsMetadata.nonKeyFieldNames) {
       final Field field = record.valueSchema().field(fieldName);
-      bindField(index++, field.schema().type(), valueStruct.get(field));
+      bindField(index++, field.schema(), valueStruct.get(field));
     }
 
     statement.addBatch();
   }
 
-  void bindField(int index, Schema.Type type, Object value) throws SQLException {
-    bindField(statement, index, type, value);
+  void bindField(int index, Schema schema, Object value) throws SQLException {
+    bindField(statement, index, schema, value);
   }
 
-  static void bindField(PreparedStatement statement, int index, Schema.Type type, Object value) throws SQLException {
+  static void bindField(PreparedStatement statement, int index, Schema schema, Object value) throws SQLException {
     if (value == null) {
       statement.setObject(index, null);
     } else {
+      if (schema.name() != null) {
+        LogicalTypeConverter logicalConverter = LOGICAL_CONVERTERS.get(schema.name());
+        if (logicalConverter != null)
+          value = logicalConverter.convert(schema, value);
```

Comment:
I think it might be better here to bind the logical types directly with 
`statement.setDate(..)`, `statement.setTime(..)`, `statement.setTimestamp(..), `statement.setBigDecimal(..)` here. But I haven't looked into this properly and this seems fine as a first pass at logical type support.